packages/kit/src/core/postbuild/prerender.js

```diff
@@ -165,9 +165,23 @@ async function prerender({ hash, out, manifest_path, metadata, verbose, env }) {
 	/**
 	 * @param {string} path
 	 * @param {boolean} is_html
+	 * @param {string} [content_type]
 	 */
-	function output_filename(path, is_html) {
-		const file = path.slice(config.paths.base.length + 1) || 'index.html';
+	function output_filename(path, is_html, content_type) {
+		const file = path.slice(config.paths.base.length + 1);
+
+		// Special handling for root path
+		if (!file) {
+			if (is_html) {
```

Comment:
Looking at how `is_html` is evaluated, I'm wondering why a redirect response automatically considers the file as a html file. We might need to rethink that one.